sinatra-contrib/lib/sinatra/namespace.rb

```diff
@@ -319,7 +319,6 @@ def compile(pattern, conditions, default_pattern = nil)
         end
         base_pattern, base_conditions = @pattern, @conditions
         pattern         ||= default_pattern
-        base_pattern    ||= base.pattern    if base.respond_to? :pattern
         base_conditions ||= base.conditions if base.respond_to? :conditions
```

Comment:
Ah, yes. However, it should be treated as a different issue.
Briefly, that `conditions` has not been permanently satisfied, so I deleted it as new commit.